connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -107,6 +110,14 @@ public Object answer() throws Throwable {
         assertEquals(200, response.getStatus());
 
         assertEquals(expectedHeader, response.getHeaderString("Access-Control-Allow-Origin"));
+
+        response = request("/connector-plugins/FileStreamSource/validate")
+            .header("Referer", origin + "/page")
+            .header("Origin", origin)
+            .header("Access-Control-Request-Method", method)
+            .options();
+        assertEquals(404, response.getStatus());
+        assertEquals(method, response.getHeaderString("Access-Control-Allow-Methods"));
```

Comment:
Should we be checking the `Access-Control-Allow-Origin` header on this request as well, just as we do with the first request in this method?